drivers/flash/flash_stm32_ospi.c

```diff
@@ -1227,6 +1403,28 @@ static int flash_stm32_ospi_write(const struct device *dev, off_t addr,
 		return 0;
 	}
 
+#ifdef CONFIG_STM32_MEMMAP
+	if (stm32_ospi_is_memorymap(dev)) {
+		/* If the Flash is in MemoryMapped mode, abort it and continue */
+		LOG_INF("MemoryMap : disable before write");
+		if (stm32_ospi_abort_memmap(dev) != 0) {
+			LOG_ERR("MemoryMap not aborted correctly");
+			return -EIO;
+		}
+	}
```

Comment:
It would be fatal when accidently writing while running in XIP.

I guess the idea here is to allow bootloader to exit memory map mode that is entered in OSPI init to allow erase & write operations.

In that case, the `CONFIG_MCUBOOT` is going to be set to `y` for the mcuboot while the app won't have it. Extra macro could be used to prevent memory map mode exit in the application code.

An alternative, protect write abortion with extra Kconfig option that the application can set.

Reply:
Another alternative is to run flash driver from RAM. Not proposed today as not yet fully functional on our side, but this is an option used by others.

Reply:
I am not sure what is the footprint on RAM in that case. But having flash driver in ram might not be required for all applications. After switching to the slot in o/qSPI it is rarely when writing and erasing are required outside of the bootloader.